jax/numpy/lax_numpy.py

```diff
@@ -545,6 +545,28 @@ def exp2(x):
   return lax.exp(lax.mul(lax.log(_constant_like(x, 2)), x))
 
 
+@_wraps(onp.signbit)
+def signbit(x):
+  x, = _promote_shapes("signbits", x)
+
+  if issubdtype(_dtype(x), onp.integer):
+    x = lax.convert_element_type(x, onp.float64)
+
+  info = finfo(_dtype(x))
```

Comment:
We should probably check for and issue a better error if a non-floating-point type makes its way here (e.g., a complex number.)